src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -527,55 +257,35 @@ public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationTo
                 return ValueTask.FromCanceled(cancellationToken);
             }
 
-            if (IsClosed)
+            if (_actualImplementation.IsClosed)
             {
                 throw Error.GetFileNotOpen();
             }
 
-            if (!_useAsyncIO)
-            {
-                // If we weren't opened for asynchronous I/O, we still call to the base implementation so that
-                // Write is invoked asynchronously.  But if we have a byte[], we can do so using the base Stream's
-                // internal helper that bypasses delegating to BeginWrite, since we already know this is FileStream
-                // rather than something derived from it and what our BeginWrite implementation is going to do.
-                return MemoryMarshal.TryGetArray(buffer, out ArraySegment<byte> segment) ?
-                    new ValueTask((Task)BeginWriteInternal(segment.Array!, segment.Offset, segment.Count, null, null, serializeAsynchronously: true, apm: false)) :
-                    base.WriteAsync(buffer, cancellationToken);
-            }
-
-            return WriteAsyncInternal(buffer, cancellationToken);
+            return _actualImplementation.WriteAsync(buffer, cancellationToken);
         }
 
         /// <summary>
         /// Clears buffers for this stream and causes any buffered data to be written to the file.
         /// </summary>
-        public override void Flush()
-        {
-            // Make sure that we call through the public virtual API
-            Flush(flushToDisk: false);
-        }
+        public override void Flush() => _actualImplementation.Flush();
 
         /// <summary>
         /// Clears buffers for this stream, and if <param name="flushToDisk"/> is true,
         /// causes any buffered data to be written to the file.
         /// </summary>
         public virtual void Flush(bool flushToDisk)
         {
-            if (IsClosed) throw Error.GetFileNotOpen();
-
-            FlushInternalBuffer();
+            if (_actualImplementation.IsClosed) throw Error.GetFileNotOpen();
```

Comment:
Here and elsewhere - wrapping in curly brackets is preffered, even if it's only one statement in the if:
```suggestion
            if (_actualImplementation.IsClosed)
            {
                throw Error.GetFileNotOpen();
            }
```

Reply:
I've tried to minimize the changes to make it easier for reviewers to focus on the actual changes. 

@carlossanlop is it OK if I address this in a separate one? I would also like to reorder the methods and properties (https://stackoverflow.com/a/310967/5852046) as this is hurting my inner pedant ;)

Reply:
Sure. Sounds good.